src/Package/Impl/Sql/Publish/SProcScriptGenerator.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.IO;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.Sql.Publish {
+    internal sealed class SProcScriptGenerator {
+        /// <summary>
+        /// Default name of the column with the stored procedure names
+        /// </summary>
+        internal const string SProcColumnName = "SProcName";
+        /// <summary>
+        /// Name of the column with the procedure code
+        /// </summary>
+        internal const string RCodeColumnName = "RCode";
+
+        private const string RCodeTemplate = "_RCODE_";
+        private const string InputQueryTemplate = "_INPUT_QUERY_";
+
+        private readonly IFileSystem _fs;
+
+        public SProcScriptGenerator(IFileSystem fs) {
+            _fs = fs;
+        }
+
+        public string CreateRCodeTableScript(SqlSProcPublishSettings settings) {
+            var sb = new StringBuilder();
+            sb.AppendLine(Invariant($"CREATE TABLE {settings.TableName.ToSqlName(settings.QuoteType)}"));
+            sb.AppendLine("(");
+            sb.AppendLine(Invariant($"{SProcColumnName} NVARCHAR(64),"));
+            sb.AppendLine(Invariant($"{RCodeColumnName} NVARCHAR(max)"));
+            sb.AppendLine(")");
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Generates SQL post deployment script that pushes R code into a table
+        /// as well as 
+        /// </summary>
+        public string CreatePostDeploymentScript(SqlSProcPublishSettings settings, IReadOnlyDictionary<string, string> sprocMap) {
```

Comment:
Do you use it anywhere as a dictionary? If it just a collection of string pairs, I'd rather add a simple type for this pair